src/state.rs

```diff
@@ -88,15 +88,28 @@ pub trait State<T, E: Send + Sync + 'static> {
     }
 
     /// Executed repeatedly at stable, predictable intervals (1/60th of a second
-    /// by default).
+    /// by default),
+    /// if this is the active state.
     fn fixed_update(&mut self, _data: StateData<T>) -> Trans<T, E> {
         Trans::None
     }
 
-    /// Executed on every frame immediately, as fast as the engine will allow (taking into account the frame rate limit).
+    /// Executed on every frame immediately, as fast as the engine will allow (taking into account the frame rate limit),
+    /// if this is the active state.
     fn update(&mut self, _data: StateData<T>) -> Trans<T, E> {
         Trans::None
     }
+
+    /// Executed repeatedly at stable, predictable intervals (1/60th of a second
+    /// by default),
+    /// even when this is not the active state,
+    /// as long as this state is on the [StateMachine](struct.StateMachine.html)'s state-stack.
+    fn shadow_fixed_update(&mut self, _data: StateData<T>) {}
+
+    /// Executed on every frame immediately, as fast as the engine will allow (taking into account the frame rate limit),
+    /// even when this is not the active state,
+    /// as long as this state is on the [StateMachine](struct.StateMachine.html)'s state-stack.
+    fn shadow_update(&mut self, _data: StateData<T>) {}
 }
 
```

Comment:
Thought: Would a shadow_handle_event make sense?
(I'm not sure myself)

Reply:
Hm, it's an interesting idea, but it might lead to problems when the shadows State and the Active state act on the same Event, or not?

Reply:
I don't think so, but I'm not sure how pertinent it is for a "paused" state to continue handling events. Probably not a lot, since events are usually used to switch between states anyway.

Reply:
I would say not, but i'm also not sure.